src/Json/JsonSchemaValidator.fs

```diff
@@ -0,0 +1,287 @@
+        {
+            MultipleOf              = tryGet "multipleOf" |> Option.bind validateMultipleOf |> Option.defaultValue 0
+            Maximum                 = tryGet "maximum" |> Option.bind validateMaximum |> Option.defaultValue 0
+            // ExclusiveMaximum     = tryGet "exclusiveMaximum" |> Option.map validateExclusiveMaximum |> Option.defaultWith 
+            // Minimum              = tryGet "minimum" |> Option.map validateMinimum |> Option.defaultWith 0
+            // MaxLength            = tryGet "maxLength" |> Option.map validateMaxLength |> Option.defaultWith 0
+            // MinLength            = tryGet "minLength" |> Option.map validateMinLength |> Option.defaultWith 0
+            // Pattern              = tryGet "pattern" |> Option.map validateExclusiveMaximum |> Option.defaultWith 0
+            Properties              = tryGet "properties" |> Option.bind validateProperties |> Option.defaultValue [||]
+            Enum                    = tryGet "enum" |> Option.bind validateEnum |> Option.defaultValue [||]
+            // Const                   = tryGet "const" |> Option.bind validateConst |> Option.defaultValue JsonValue.Null
+            Type                    = tryGet "type" |> Option.bind validateType |> Option.defaultValue [||]
+            AllOf                   = tryGet "allOf" |> Option.bind validateAllOf |> Option.defaultValue [||]
+            AnyOf                   = tryGet "anyOf" |> Option.bind validateAnyOf |> Option.defaultValue [||]
+            Title                   = tryGet "title" |> Option.bind (function | JsonValue.String name -> Some name | _ -> None) |> Option.defaultValue "INVALID"
+        } |> Some
+    | _ -> None
+
+and validateItems value =
+
+    None
+
+and validateAdditionalItems value =
+
+    None
+
+and validateMaxItems value =
+
+    None
+
+and validateMinItems value =
+
+    None
+
+and validateUniqueItems value =
+
+    None
+
+and validateContains value =
+
+    None
+
+and validateMaxProperties value =
+
+    None
+
+and validateMinProperties value : decimal option =
+
+    None
+
+and validateRequired value : float option =
+
+    None
+
+and validateProperties value : JsonSchema [] option =
+    match value with
+    | JsonValue.Record properties ->
+        [| for property in properties do
+               let name, value = property
+               let res = validateRoot value |> Option.get
+               yield { res with Title = name } |] |> Some
+    | _ -> None
+
+and validatePatternProperties value =
+
+    None
+
+and validateAdditionalProperties value =
+
+    None
+
+and validateDependencies value =
+
+    None
+
+and validatePropertyNames value =
+
+    None
+
+and validateEnum value : JsonValue [] option =
+    match value with
+    | JsonValue.Array content -> Some content
+    | _ -> None
+
+and validateConst value =
+
+    None
+
+and validateType value : JsonValue [] option =
```

Comment:
Consider adding a `JsonValueType` record which concentrates focus on the type, not the values.